playbooks/robusta_playbooks/job_restart_on_oomkilled.py

```diff
@@ -0,0 +1,247 @@
+from robusta.api import (
+    action,
+    ActionParams,
+    Optional,
+    JobEvent,
+    Finding,
+    Container,
+    FindingSource,
+    logging,
+    FindingType,
+    MarkdownBlock,
+    ResourceRequirements,
+    RobustaJob,
+    ObjectMeta,
+    JobSpec,
+    PodSpec,
+    PodTemplateSpec,
+)
+from ...src.robusta.core.playbooks.job_utils import get_job_latest_pod
+import bitmath as bitmath
+
+
+class IncreaseResources(ActionParams):
+
+    """
+    :var increase_by: (optional).Users will specify how much they want to increase in each restart, (e.g: 500MiB).
+    :var max_resource: This variable prevent an infinite loop of job's pod crashing and getting more memory.The action won't increase the memory again when the "Max" limit reached, (eg: 1GiB).
+    """
+
+    increase_by: Optional[str] = 0
+    max_resource: str
+
+
+@action
+def job_restart_on_oomkilled(event: JobEvent, params: IncreaseResources):
+    """
+    This action will run when job failed with oomkilled
+    """
+    function_name = "job_restart_on_oomkilled"
+    finding = Finding(
+        title=f"JOB RESTARTED",
+        source=FindingSource.MANUAL,
+        aggregation_key=function_name,
+        finding_type=FindingType.REPORT,
+        failure=False,
+    )
+    if not event.get_job():
+        logging.error(f"job restart was called on event without job: {event}")
+        return
+    job_event = event.get_job()
+
+    """
+    Retrieves job's pod information
+    """
+    try:
+        pod = get_job_latest_pod(job_event)
+
+    except:
+        logging.error(f"get_job_pod was called on event without job: {event}")
+        return
+
+    containers = []
+    oomkilled_containers = []
+    running_containers = []
+
+    """
+    Retrieves pod's containers information
+    """
+    OOMKilled = "OOMKilled"
+    for status in pod.status.containerStatuses:
+        if status.state.running == None:
+            if status.state.terminated.reason == OOMKilled:
+                oomkilled_containers.append(status.name)
+        else:
+            running_containers.append(status.name)
+
+    """
+    Updating pod's containers resources if required
+    """
+    for index, container in enumerate(pod.spec.containers):
+        if container.name in oomkilled_containers:
+
+            req_memory = bitmath.parse_string_unsafe(
```

Comment:
if an oomkilled_container does not have requests or limit, I think we should fail the actions immediately and not restart the job

Reply:
resolved.